Assets/ModScripts/Submodules/StumblingSymphony.cs

```diff
@@ -0,0 +1,221 @@
+            }
+            finalInput.Add(noteSeq[i]);
+        }
+    } 
+
+    int FindStumbleNote(int initialNote)
+    {
+        int[] shifter = { 1, 8, 7, 5, 6, 3, 10, 9, 2, 11, 4 };
+        semiShift = shifter[Info.Button];
+        initialNote = (initialNote + semiShift) % 12;
+        return initialNote;
+    }
+
+    public override void OnUtilityPress()
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.UtilityButton.GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.5f);
+
+        if (Module.IsModuleSolved() || Module.IsSolving())
+            return;
+
+        if (!Module.CheckValidComponents())
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The ❖ button was pressed when the component selection was [{1}] instead of [{2}].", ModuleID, Module.GetOnComponents(), Module.GetTargetComponents());
+            Module.CauseStrike();
+            return;
+        }
+
+        Module.StartSolve();
+        PickNumber();
+        Debug.LogFormat("[The Cruel Modkit #{0}] The melody is {1}.", ModuleID, melodyNames[Info.NumberDisplay]);
+        Debug.LogFormat("[The Cruel Modkit #{0}] The rule used is rule {1}. {2}", ModuleID, activeRule + 1, ruleLog[activeRule]);
+        CalcInputs();
+        Debug.LogFormat("[The Cruel Modkit #{0}] The full sequence of notes to play (including stumbles) is {1}.", ModuleID, string.Join(", ", finalInput.Select(x => noteOrder[x]).ToArray()));
+        Debug.LogFormat("[The Cruel Modkit #{0}] The button is {1}. Your stumbles will be shifted {2} semitones forward", ModuleID, Info.MainColors[Info.Button], semiShift);
+        return;
+    }
+
+    public override void OnPianoPress(int Piano)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlaySoundAtTransform(Module.PianoSounds[Piano + (Info.Piano * 12)].name, Module.transform);
+        Module.Piano[Piano].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved() || !Module.IsSolving())
+            return;
+        
+
+        if (Piano == finalInput[numInputs])
+            numInputs++;
+        else
+        {
+            Module.CauseStrike();
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! You played {1} when {2} was needed.", ModuleID, noteOrder[Piano], noteOrder[finalInput[numInputs]]);
+            numInputs = 0;
+            return;
+        }
+
+        if (buttonPresses.Contains(numInputs - 1) && !buttonPressed)
+        {
+            Module.CauseStrike();
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! You have forgotten to press the button after a stumble (note {1})!", ModuleID, noteOrder[finalInput[numInputs-2]]);
+            numInputs = 0;
+            return;
+        }
+        else if (buttonPresses.Contains(numInputs - 1) && buttonPressed)
```

Comment:
```suggestion
        else if (buttonPresses.Contains(numInputs - 1))
```
Remove unnecessary condition. buttonPressed can't be false here, otherwise that would fall under the previous condition